MapboxMobileEvents/MMEEvent.m

```diff
@@ -253,36 +275,100 @@ + (NSString *)deviceOrientation {
     }
     return result;
 }
+#endif
+
+#pragma mark - NSSecureCoding
+
++ (BOOL) supportsSecureCoding {
+    return YES;
+}
+
+#pragma mark -
+
+- (instancetype) initWithDate:(MMEDate *)eventDate name:(NSString *)eventName  attributes:(NSDictionary *)eventAttributes {
+    if (self = [super init]) {
+        _date = eventDate;
+        _name = eventName;
+        _attributes = eventAttributes;
+    }
+
+    return self;
+}
 
 - (BOOL)isEqualToEvent:(MMEEvent *)event {
     if (!event) {
         return NO;
     }
     
     BOOL hasEqualName = [self.name isEqualToString:event.name];
+    BOOL hasEqualDate = (self.date.timeIntervalSinceReferenceDate == event.date.timeIntervalSinceReferenceDate);
     BOOL hasEqualAttributes = [self.attributes isEqual:event.attributes];
     
-    return hasEqualName && hasEqualAttributes;
+    return (hasEqualName && hasEqualDate && hasEqualAttributes);
 }
 
+#pragma mark - NSObject overrides
+
 - (BOOL)isEqual:(id)other {
     if (other == self) {
         return YES;
     }
     
-    if (![other isKindOfClass:[MMEEvent class]]) {
+    if (![other isKindOfClass:MMEEvent.class]) {
         return  NO;
     }
     
     return [self isEqualToEvent:other];
 }
 
 - (NSUInteger)hash {
-    return self.name.hash ^ self.attributes.hash;
+    return (self.name.hash ^ self.attributes.hash);
 }
 
 - (NSString *)description {
-    return [NSString stringWithFormat:@"<%@ name=%@, attributes=%@>", NSStringFromClass([self class]), self.name, self.attributes];
+    return [NSString stringWithFormat:@"<%@ name=%@, date=%@, attributes=%@>",
+        NSStringFromClass(self.class), self.name, self.date, self.attributes];
+}
+
+#pragma mark - NSCopying
+
+- (id)copyWithZone:(NSZone *)zone {
+    MMEEvent *copy = [MMEEvent new];
+    copy.name = [self.name copy];
```

Comment:
The properties are already marked as `copy`, so the manual copy is superfluous here.